components/payments/cmd/connectors/internal/connectors/modulr/task_fetch_accounts.go

```diff
@@ -51,37 +73,42 @@ func fetchAccounts(
 	connectorID models.ConnectorID,
 	ingester ingestion.Ingester,
 	scheduler task.Scheduler,
-) error {
+	state fetchAccountsState,
+) (fetchAccountsState, error) {
 	for page := 0; ; page++ {
-		pagedAccounts, err := client.GetAccounts(ctx, page, config.PageSize)
+		pagedAccounts, err := client.GetAccounts(ctx, page, config.PageSize, state.LastCreatedDate)
 		if err != nil {
-			return err
+			return state, err
 		}
 
 		if len(pagedAccounts.Content) == 0 {
 			break
 		}
 
 		if err := ingestAccountsBatch(ctx, connectorID, ingester, pagedAccounts.Content); err != nil {
-			return err
+			return state, err
 		}
 
 		for _, account := range pagedAccounts.Content {
+			if err := state.SetLatest(account); err != nil {
```

Comment:
Hoe do you make sure that accounts are sorted, so the the set latest always sets the last created at ?

Reply:
My bad I didn't see the sortFields and sortOrder